### PR TITLE
ARROW-12415: [CI] [Python] Failed building wheel for pygit2 on ARM64 

### PR DESCRIPTION
**Pull request by pachadotdev**

### CODE REVIEW

**Comment by pachamaltese** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-arm64

**Comment by kszucs** [comment]
Thanks @pachamaltese for working on this! 

Could you please try to install pygit2 from conda at https://github.com/apache/arrow/blob/0955e547633169acbb9ac23d9966d0a2abe8d386/dev/tasks/conda-recipes/drone-steps.sh#L17

? Hopefully conda ships precompiled binaries for pygit2 on arm64.

**Comment by pachamaltese** [comment]
> Thanks @pachamaltese for working on this!
> 
> Could you please try to install pygit2 from conda at
> 
> https://github.com/apache/arrow/blob/0955e547633169acbb9ac23d9966d0a2abe8d386/dev/tasks/conda-recipes/drone-steps.sh#L17
> 
> ? Hopefully conda ships precompiled binaries for pygit2 on arm64.

ok, ready and pushed, now I'll test

**Comment by pachamaltese** [comment]
@kszucs hi, it should be ready by now, at least it passes the CI tests but tonight we shall be sure

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-arm64

**Review by kszucs — APPROVED** [review]
LGTM, the crossbow task should verify that it indeed works.

**Comment by pachamaltese** [comment]
> LGTM, the crossbow task should verify that it indeed works.

hi
it still fails with "ERROR: Failed building wheel for pygit2"
https://cloud.drone.io/ursacomputing/crossbow/563/1/2

**Comment by kszucs** [comment]
Strange, pygit2 should be installed by conda at that step. Triggering another build to inspect the environment.

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-arm64

**Comment by kszucs** [comment]
There is a strange error which shouldn't occur - at least I'm unable to reproduce locally - so debugging remotely.

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-arm64

**Comment by kszucs** [comment]
Since crossbow is an optional subpackage the import error has been swallowed.

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-gcc-py36-arm64

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-*-arm64

**Comment by kszucs** [comment]
Finally, things should work now.

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-linux-*-arm64

**Comment by kszucs** [comment]
Some of the builds have timeouts (just barely crossing 60m), but in a non-deterministic fashion. Perhaps drone.io schedules them on different machines.

**Comment by kszucs** [comment]
@xhochy what are your plans for the aarch64 builds? Shall we move to virtualized builds on azure pipelines?

**Review by kszucs — APPROVED** [review]
The immediate artifact deployment issue have been resolved.
We still need to handle the build timeouts (unless we move to another CI), but we can handle that in a follow-up.

File: dev/archery/archery/cli.py [hunk 1]
```diff
@@ -1076,7 +1076,10 @@ def release_cherry_pick(obj, version, dry_run, recreate):
 
     @archery.command(
         'crossbow',
-        context_settings={"ignore_unknown_options": True}
+        context_settings={
+            "allow_extra_args": True,
```
Comment (kszucs) [inline dev/archery/archery/cli.py:1080]: Needed for proper error reporting when trying to call the optional `archery crossbow` command without the required dependencies installed.

**Comment by xhochy** [comment]
> @xhochy what are your plans for the aarch64 builds? Shall we move to virtualized builds on azure pipelines?

We should move to *cross-compiled* builds on Azure.

**Comment by kszucs** [comment]
Ohh, cross-compiling sounds nice!

**Pull request closed without merging**